task/rsync/0.1/README.md

```diff
@@ -0,0 +1,237 @@
+# Rsync
+
+This task provides synchronization of files between container and remote host. It relies on rsync and sshpass commands to implement.
+
+## Install the Task
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/rsync/0.1/rsync.yaml
+```
+
+## Parameters
+
+- **direction**: "local-to-remote" means synchronization from the container to the remote host, and "remote-to-local" means synchronization from the remote host to the container. (_default:_ "local-to-remote")
+- **local-path**: The directory in the container that you want to synchronize (_default:_ "./")
+- **remote-ip**: IP of the remote host (_default:_ "0.0.0.0").
+- **remote-username**: Username of the remote host. (_default:_ "root")
+- **remote-password**: Password of the remote host. (_default:_ "password")
+- **remote-path**: The directory in the remote host that you want to synchronize (_default:_ "~/")
+
+## Usage
+
+- Install rsync and sshpass in remote host
+
+```bash
+yum install -y rsync sshpass
+```
+
+- Create the `PipelineRun`
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: rsync-remote-to-local
+spec:
+  pipelineSpec:
+    workspaces:
+      - name: workspace
+    tasks:
+      - name: rsync-files
+        taskRef:
+          name: rsync
+        params:
+          - name: direction
+            value: "remote-to-local"
+          - name: local-path
+            value: "."
+          - name: remote-ip
+            value: "0.0.0.0"
+          - name: remote-username
+            value: "root"
+          - name: remote-password
+            value: "mypassword"
+          - name: remote-path
+            value: /root
+        workspaces:
+          - name: source
+            workspace: workspace
+      - name: verify
+        taskSpec:
+          workspaces:
+            - name: source
+          steps:
+            - name: shell
+              image: alpine
+              workingDir: $(workspaces.source.path)
+              script: |
+                #!/bin/sh
+                pwd
+                ls -al
+        workspaces:
+          - name: source
+            workspace: workspace
+        runAfter:
+          - rsync-files
+  workspaces:
+    - name: workspace
+      volumeClaimTemplate:
+        spec:
+          accessModes:
+            - ReadWriteOnce
+          resources:
+            requests:
+              storage: 100M
+```
+
+
+- View remote host directory
+
+```bash
+[root@myk8s ~]# ls -al /root/
+
```

Comment:
I don't think we need to copy the whole outputs in there.

Reply:
Do you have any better suggestions? Here is just an example.

Reply:
well, you can just omit the output.

Reply:
OK